src/Microsoft.SqlTools.ResourceProvider.Core/Firewall/FirewallRuleService.cs

```diff
@@ -55,31 +56,40 @@ public class FirewallRuleService : IFirewallRuleService
         /// <summary>
         /// Creates firewall rule for given server name and IP address range. Throws exception if operation fails
         /// </summary>
-        public async Task<FirewallRuleResponse> CreateFirewallRuleAsync(string serverName, string startIpAddressValue, string endIpAddressValue)
+        public async Task<FirewallRuleResponse> CreateFirewallRuleAsync(CreateFirewallRuleParams firewallRuleParams)
         {
-            IPAddress startIpAddress = ConvertToIpAddress(startIpAddressValue);
-            IPAddress endIpAddress = ConvertToIpAddress(endIpAddressValue);
-            return await CreateFirewallRuleAsync(serverName, startIpAddress, endIpAddress);
+            IPAddress startIpAddress = ConvertToIpAddress(firewallRuleParams.StartIpAddress);
+            IPAddress endIpAddress = ConvertToIpAddress(firewallRuleParams.EndIpAddress);
+            FirewallRuleRequest firewallRuleRequest = new FirewallRuleRequest()
+            {
+                StartIpAddress = ConvertToIpAddress(firewallRuleParams.StartIpAddress),
+                EndIpAddress = ConvertToIpAddress(firewallRuleParams.EndIpAddress),
+                FirewallRuleName = string.Format(CultureInfo.InvariantCulture, firewallRuleParams.FirewallRuleName ?? "ClientIPAddress_{0}",
```

Comment:
You shouldn't be formatting the passed in FirewallRuleName, right? Or if it's intentional that you'll be inserting the date you should document that in the API.